### PR TITLE
trace output handlers & event consumers

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java [hunk 1]
```diff
@@ -40,15 +41,14 @@
...
-import java.util.function.BiConsumer;
 import javaslang.control.Try;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * An event consumer that integrates {@link RunState.State} values with a {@link Publisher}.
  */
-public class PublisherHandler implements BiConsumer<SequenceEvent, RunState> {
+public class PublisherHandler implements StyxScheduler.EventConsumer {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java]: This however creates a circular dependency. How about we make `StyxScheduler.EventConsumer` a proper interface?

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/TransitionLogger.java [hunk 2]
```diff
@@ -23,16 +23,16 @@
...
 import com.spotify.styx.state.RunState;
 import com.spotify.styx.util.EventUtil;
 import java.util.Locale;
 import java.util.Objects;
-import java.util.function.BiConsumer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class TransitionLogger implements BiConsumer<SequenceEvent, RunState> {
+public class TransitionLogger implements StyxScheduler.EventConsumer {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/TransitionLogger.java]: Same here.

Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/TransitionLogger.java]: 👍

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java (hunk 1 above)
Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java]: 👍

**Comment by danielnorberg** [comment]
👍

**Pull request merged**